setup.py

```diff
@@ -55,12 +54,219 @@ def build_extensions(self):
+                ],
+                include_dirs=['.', 'pygsti/evotypes', np.get_include()],
+                language="c++",
+                extra_link_args=["-std=c++11"]
+            ),
+            Extension(
+                "pygsti.evotypes.stabilizer.termreps",
+                sources=[
+                    "pygsti/evotypes/stabilizer/termreps.pyx",
+                    "pygsti/evotypes/stabilizer/termcreps.cpp",
+                    "pygsti/evotypes/stabilizer/statecreps.cpp",
+                    "pygsti/evotypes/stabilizer/opcreps.cpp",
+                    "pygsti/evotypes/stabilizer/effectcreps.cpp"
+                ],
+                include_dirs=['.', 'pygsti/evotypes', np.get_include()],
+                language="c++",
+                extra_link_args=["-std=c++11"]
+            ),
+            Extension(
+                "pygsti.forwardsims.mapforwardsim_calc_densitymx",
+                sources=[
+                    "pygsti/forwardsims/mapforwardsim_calc_densitymx.pyx",
+                    "pygsti/evotypes/densitymx/statecreps.cpp",
+                ],
+                include_dirs=['.', 'pygsti/evotypes', 'pygsti/evotypes/densitymx', np.get_include()],
+                language="c++",
+                extra_link_args=["-std=c++11"]
+            ),
+            Extension(
+                "pygsti.forwardsims.termforwardsim_calc_statevec",
+                sources=[
+                    "pygsti/forwardsims/termforwardsim_calc_statevec.pyx",
+                    "pygsti/evotypes/statevec/statecreps.cpp",
+                    "pygsti/evotypes/basecreps.cpp"
+                ],
+                include_dirs=['.', 'pygsti/evotypes', 'pygsti/evotypes/statevec', np.get_include()],
+                language="c++",
+                extra_link_args=["-std=c++11"]
+            ),
+            Extension(
+                "pygsti.forwardsims.termforwardsim_calc_stabilizer",
+                sources=[
+                    "pygsti/forwardsims/termforwardsim_calc_stabilizer.pyx",
+                    "pygsti/evotypes/stabilizer/statecreps.cpp",
+                    "pygsti/evotypes/basecreps.cpp"
+                ],
+                include_dirs=['.', 'pygsti/evotypes', 'pygsti/evotypes/stabilizer', np.get_include()],
+                language="c++",
+                extra_link_args=["-std=c++11"]
+            ),
+            Extension(
+                "pygsti.circuits.circuitparser.fastcircuitparser",
+                sources=["pygsti/circuits/circuitparser/fastcircuitparser.pyx"],
+                include_dirs=['.', np.get_include()],
+                language="c++",
+                extra_link_args=["-std=c++11"]
+            )
+        ]
+        extensions = cythonize(ext_modules, compiler_directives={'language_level': "3"}, exclude_failures="PYGSTI_CYTHON_EXCLUDE_FAILURES" in os.environ)
+except ImportError:
+    warn("Extensions build tools are not available. Installing without Cython extensions...")
+    extensions = None
```

Comment:
We still don't build extensions if no Cython. However, as mentioned in my main message, **this should basically never trigger** because it has to do with the build-system requirements, not the user-specified optional dependencies.